### ISSUE TITLE
implicit_getter false positive when there's getter and setter backed by an Obj-C associated object

### ISSUE THREAD

**Issue by rogerluan** [realm/SwiftLint#3222]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

The `implicit_getter` rule is triggering a warning when my property is a computed variable with getter and setter. This is happening with all my computed variables with getters and setters **that use Obj-C Associated Objects as a backvar**. All of these use cases are inside `extensions` in my project, and the warning only triggers when the extension is non-private. See code snippet below for an example.

##### Complete output when running SwiftLint, including the stack trace and command used

It doesn't trigger the warning via CLI.

### Environment

* SwiftLint version 0.39.2
* Installation method used: CocoaPods
* Paste your configuration file: Nothing fancy, this happens using default rules.
* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)? No.
* Which Xcode version are you using (check `xcodebuild -version`)? Xcode 11.5, Build version 11E608c
* Do you have a sample that shows the issue? Running via CLI doesn't trigger the warning for some reason.

```swift
// This triggers a violation:
extension UIView {
    private enum Association {
        static var persistentAnimationsKey = 0
    }

    var persistentAnimations: [String:CAAnimation] {
        get { return objc_getAssociatedObject(self, &Association.persistentAnimationsKey) as? [String:CAAnimation] ?? [:] }
        set { objc_setAssociatedObject(self, &Association.persistentAnimationsKey, newValue, objc_AssociationPolicy.OBJC_ASSOCIATION_RETAIN_NONATOMIC) }
    }
}

// This doesn't trigger a violation:
private extension UIView {
    private enum Association {
        static var persistentAnimationsKey = 0
    }

    var persistentAnimations: [String:CAAnimation] {
        get { return objc_getAssociatedObject(self, &Association.persistentAnimationsKey) as? [String:CAAnimation] ?? [:] }
        set { objc_setAssociatedObject(self, &Association.persistentAnimationsKey, newValue, objc_AssociationPolicy.OBJC_ASSOCIATION_RETAIN_NONATOMIC) }
    }
}

// The only difference between the first and the second is the presence of the `private` key on the extension.
```

**Comment by marcelofabri** [comment]
Are you sure you're using 0.39.2? I couldn't reproduce it and this looks a lot like #3149

**Comment by rogerluan** [comment]
@marcelofabri you're right, my bad! 

I had upgraded SwiftLint pod to 0.39.2 but it got reversed in a bad Podfile.lock merge, and then I ran `swiftlint version` which obviously doesn't look up the Pod 😓 

Thanks for helping me spot this! 🚀 I confirmed that the issue was indeed fixed by upgrading 👍
